hooks/pre-commit

```diff
@@ -15,12 +15,34 @@
 gofiles=$(git diff --cached --name-only --diff-filter=ACM | grep '.go$')
 [ -z "$gofiles" ] && exit 0
 
+govet=$(go vet $gofiles)
+retcode=$?
+if [ ! -z "$govet" ]; then
+	echo "go vet checks not passed: Exit code $retcode \n"
```

Comment:
My mistake for misleading you in my previous review. If we do this in a situation where I only changed a single file that uses values from another -- for example, if I change `db/db.go` but don't change `db/constants.go`, then `go vet` won't know what to do.

![image](https://user-images.githubusercontent.com/7704542/75013232-a288d680-5438-11ea-97fc-942331778d76.png)

This does bring up another point, though, in that my commit still passed despite the clear output. I would advise that you check the return code from the command here instead of the output of the command. This also applies to when checking if `golint` returned error-free on L27. Aside from this the code looks good ᵉˣᶜᵉᵖᵗ ᶠᵒʳ ᵗʰᵃᵗ ᵉˣᵗʳᵃ ˢᵖᵃᶜᵉ ᵒⁿ ᴸ²¹. Thanks again Alyssa!